File: Kaviyarasan/June-26/pressed_flower/update_mobile_nav.py
import os, re
import glob

# Files
files = glob.glob('d:/Projects/Templates/Kaviyarasan/June-26/pressed_flower/*.html') + glob.glob('d:/Projects/Templates/Kaviyarasan/June-26/pressed_flower/pages/*.html')

for filepath in files:
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # 1. Remove SVGs inside mobile-nav-link
    new_content = re.sub(r'<a href="([^"]+)" class="mobile-nav-link([^"]*)">\s*<svg.*?</svg>\s*(.*?)\s*</a>', r'<a href="\1" class="mobile-nav-link\2">\3</a>', content, flags=re.DOTALL)
    
    # 2. Update texts
    new_content = new_content.replace('>About Us</a>', '>About</a>')
    new_content = new_content.replace('>Services & Custom orders</a>', '>Service</a>')
    new_content = new_content.replace('>Pricing Tiers</a>', '>Pricing</a>')
    new_content = new_content.replace('>The Journal</a>', '>Blog</a>')
    new_content = new_content.replace('>Contact Studio</a>', '>Contact</a>')
    
    # 3. Remove Other Links section
    new_content = re.sub(r'<div class="nav-section">\s*<div class="nav-section-label">Other Links</div>.*?</div>\s*</div>\s*</nav>', r'</div>\n  </nav>', new_content, flags=re.DOTALL)

    # 4. Remove Studio Pages label
    new_content = re.sub(r'<div class="nav-section-label">Studio Pages</div>\s*', r'', new_content)

    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(new_content)
    print(f'Updated {filepath}')

File: Kaviyarasan/June-26/pressed_flower/update_mobile_nav.js
const fs = require('fs');

// Use native fs functions to glob since glob might not be installed
const path = require('path');

function getFiles(dir, filter) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            if (!file.includes('assets')) {
               results = results.concat(getFiles(file, filter));
            }
        } else {
            if (file.endsWith(filter)) results.push(file);
        }
    });
    return results;
}

const files = getFiles('d:/Projects/Templates/Kaviyarasan/June-26/pressed_flower', '.html');

files.forEach(filepath => {
    let content = fs.readFileSync(filepath, 'utf8');

    // 1. Remove SVGs inside mobile-nav-link
    content = content.replace(/<a href="([^"]+)" class="mobile-nav-link([^"]*)">\s*<svg[\s\S]*?<\/svg>\s*(.*?)\s*<\/a>/g, '<a href="$1" class="mobile-nav-link$2">$3</a>');
    
    // 2. Update texts
    content = content.replace(/>About Us<\/a>/g, '>About</a>');
    content = content.replace(/>Services & Custom orders<\/a>/g, '>Service</a>');
    content = content.replace(/>Pricing Tiers<\/a>/g, '>Pricing</a>');
    content = content.replace(/>The Journal<\/a>/g, '>Blog</a>');
    content = content.replace(/>Contact Studio<\/a>/g, '>Contact</a>');
    
    // 3. Remove Other Links section
    content = content.replace(/<div class="nav-section">\s*<div class="nav-section-label">Other Links<\/div>[\s\S]*?<\/div>\s*<\/div>\s*<\/nav>/g, '</div>\n  </nav>');

    // 4. Remove Studio Pages label
    content = content.replace(/<div class="nav-section-label">Studio Pages<\/div>\s*/g, '');

    fs.writeFileSync(filepath, content, 'utf8');
    console.log(`Updated ${filepath}`);
});
